### DISCUSSION TITLE
Recognition returns extra characters

### DISCUSSION THREAD

**Discussion by richardgohth** [PaddlePaddle/PaddleOCR#13284] [Q&A]
I am training paddleocr on synthetic characters.
However always get extra characters during recognition
For example
1234 is read as 123400

Anyone encountered? How to resolve?
Appreciate assistance please

**Comment by UserWangZz** [comment]
Can you provide an example of an incorrectly recognized image?

**Reply by richardgohth** [reply]
Thanks for quick response.

![syn_arabic_0025](https://github.com/PaddlePaddle/PaddleOCR/assets/60122996/8315452d-e237-4916-aae7-2d9c84d241af)
This image is labelled as 0025 but recognised as 002500
====
[rec_arabic_train.txt](https://github.com/user-attachments/files/16123767/rec_arabic_train.txt)

**Reply by UserWangZz** [reply]
1. How much data is there in the training set and how much data is in the validation set? 
2. From your config file, it seems that you are trying to use large epochs to promote model fitting. Does this mean that the amount of your training data is not large? 
3. Also seeing that your validation set seems to use the icdar2015 data set, please ensure that your training set and validation set come from the same distribution as much as possible.

**Reply by richardgohth** [reply]
Thanks for your review
i dont have a validation set. just testing against the training set to see if the training can converge. 
The validation in the config file is ignored.
the config yml is copied from githuh. the training set is 99k , synthetic. all 4 digits. 
Anything I should/can change in config file? I am new to paddleocr rec training, thanks

**Reply by UserWangZz** [reply]
1. Check whether the first 4 recognition results are correct. If so, it is recommended to use post-processing to get the correct answer, such as only taking the first 4 predicted characters.
2. If it is not correct, it is recommended to generate some data of variable length to see if this problem can be solved.

**Reply by richardgohth** [reply]
Thanks.
Will try as adviced.
Some questions
Have used perfect training and perfect (same) test set, why should the training not converge considering the pre-trained recognition model performs better then 99.999 %? 
At worst training shld be overfitting.
iow, why is this happening?
am i doing something fundamentally wrong?

I did try before, training on vertical characters same thing happened, ie extra characters.
Thanks for advise.
